utils/single_cell_utils.py

```diff
@@ -0,0 +1,266 @@
+    phenotypic_classes: list,
+    dataset: pd.DataFrame,
+    log_reg_model: LogisticRegression,
+    single_cell_images_dir_path: pathlib.Path,
+) -> pd.DataFrame:
+    """
+    get 15 correct sample images, 3 for each 5 phenotypic classes given as inputs
+
+    Parameters
+    ----------
+    phenotypic_classes : list
+        list of phenotypic classes to get 3 correct sample images for
+    dataset : pd.DataFrame
+        cell dataset with metadata and features
+    log_reg_model : LogisticRegression
+        model used to classify cells
+    single_cell_images_dir_path : pathlib.Path
+        path to single cell sample images
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with phenotypic classes and paths to 3 sample images that the model correctly predicted
+    """
+    compiled_sample_images = []
+
+    for phenotypic_class in phenotypic_classes:
+        # get images of desired phenotypic class
+        sample_image_paths = get_class_sample_images(
+            phenotypic_class,
+            dataset,
+            log_reg_model,
+            single_cell_images_dir_path,
+            correct=True,
+        )
+        # add these paths to compiled sample images
+        sample_image_paths_dataframe = pd.Series(sample_image_paths).to_frame().T
+        compiled_sample_images.append(sample_image_paths_dataframe)
+
+    # convert compiled sample images to dataframe and add prefix to column names
+    compiled_sample_images = pd.concat(compiled_sample_images).add_prefix("Path_")
+    # add phenotypic classes to compiled sampel images dataframe
+    compiled_sample_images.insert(
+        loc=0, column="Phenotypic_Class", value=phenotypic_classes
+    )
+
+    return compiled_sample_images.reset_index(drop=True)
+
+
+def plot_15_correct_sample_images(sample_images_df: pd.DataFrame):
+    """
+    show the 15 correct sample images collected with get_15_correct_sample_images()
+
+    Parameters
+    ----------
+    sample_images_df : pd.DataFrame
+        dataframe with phenotypic classes and paths to 3 sample images that the model correctly predicted
+    """
+    path_columns = [column for column in sample_images_df.columns if "Path_" in column]
+
+    # create 3x5 plot for plotting images
+    fig, axs = plt.subplots(3, 5)
+    fig.set_size_inches(14, 8)
+    ax_x = 0
+    ax_y = 0
+
+    # iterate through sample images data to plot each class sample images
+    for _, sample_images_data in sample_images_df.iterrows():
+        paths = sample_images_data[path_columns].to_list()
+
+        for path in paths:
+            sample_image = mpimg.imread(path)
+
+            axs[ax_x, ax_y].imshow(sample_image, cmap="gray")
+
+            # only add pehnotypic class to images at the top of the figure
+            if ax_x == 0:
+                axs[ax_x, ax_y].set_title(sample_images_data["Phenotypic_Class"])
+
+            ax_x += 1
+            if ax_x == 3:
+                ax_x = 0
+                ax_y += 1
```

Comment:
Consider providing a comment or two here to explain what's happening.

Reply:
Done in https://github.com/WayScience/phenotypic_profiling_model/pull/10/commits/f7d7ac058baed37827161535af842c9953b6aa1c